### ISSUE TITLE
Can't find this document

### ISSUE THREAD

**Issue by vnckppl** [mmagnus/emacs-grammarly#10]
Is emacs-grammarly broken because of changes in the Grammarly app? I am on Grammarly 1.5.64 and when I run `grammarly-save-region-and-run` on a region, Grammarly opens but shows me the error message 'Can't find this document'.

**Comment by mmagnus** [comment]
thanks, @vnckppl Yeah, now I see the problem as well.. hmm.. I have to read more about.

One workout would be to use AppleScript to open Grammarly, to "click" Edit and Paste. 

This actually could be interesting workout, because based on that I could build a way to copy the Grammarly content back to the Emacs.. so maybe that Grammarly broke the plugin will be good for us :) Still I need some time to code it.

A quick and dirty workout is to downgrade Grammarly :( and wait for an update here (a few days).

<img width="700" alt="Screen Shot 2020-04-16 at 11 57 56 AM" src="https://user-images.githubusercontent.com/118740/79443282-eda70c00-7fd9-11ea-9af6-15b1d3587df5.png">

**Comment by vnckppl** [comment]
The Applescript option would be cool, although it seems you have to start a new document before you can paste, and there is no 'new document' menu option. 

Finding an older version of Grammarly is challenging. I will let you know when I find a download link.

**Comment by mmagnus** [comment]
Yeah, correct, you need to start a new document. I'm thinking if there is some way around....

**Comment by agzam** [comment]
Okay, folks. It seems I'm not the only one struggling here with this Grammarly bullshit. The level of imbecilic "vision" of their UX experts is simply overwhelming. I understand why they wouldn't even try to build an API for the service. But changing the app so it is extremely painful to use, even for the paid users, makes no sense whatsoever.
I spent almost an hour, trying to come up with a workaround. Even went into exploring the app with the dev console:
`open -a //Applications/Grammarly.app/contents/macos/Grammarly --args -o`.

Yet I couldn't come up with any better idea, except keeping the app open of all time, with the focus on the same document - to use it as a "whiteboard." And maybe Apple Script to activate that window, and paste into it; and then clearing it up after, but never to close it.

If I'd to make it work, I would probably use Hammerspoon to introduce a keyboard watcher, for "done with the checking - want to go back to Emacs" thing, I'm not sure how to make this work with Apple Script.

**Comment by mmagnus** [comment]
OK, for now, I found in my backup an old version of Grammarly. I renamed so it will not be overwritten the new Grammarly. This is some workout for now. This is very annoying since this Grammarly will bug you with updating it.

<img width="1169" alt="Screen Shot 2020-04-20 at 11 38 39 AM" src="https://user-images.githubusercontent.com/118740/79737350-7df79080-82fb-11ea-86ba-a6d646e9d195.png">

**Comment by mmagnus** [comment]
OK, this is even crazier. You can open this file I put here but it will update automatically to the newest version.. grr..  

<img width="178" alt="Screen Shot 2020-04-20 at 11 52 33 AM" src="https://user-images.githubusercontent.com/118740/79738731-71743780-82fd-11ea-865f-d3fb587be65c.png">
I open "my" grammarly a few times with ignoring updates, but still, after a while the app is updated...

**Comment by mmagnus** [comment]
BTW, rename the zip Grammarly to Grammarly_61 in your Applications

**Comment by vnckppl** [comment]
It does indeed automatically update. One way of preventing is, is to block Grammarly from phoning home, but that would just lead us further down the rabbit hole.

**Comment by mmagnus** [comment]
OK, I just asked Grammarly if they can help and give us this basic funcionality :-) we'll see..

**Comment by mmagnus** [comment]
> Hi Marcin,
> 
> Thanks for contacting Grammarly!
> 
> Unfortunately, we don’t currently have a quick fix, but please rest assured that our team has been made aware of the issue. I can’t provide you with an estimated timeframe right now because the developers need some time to analyze the cause of the issue.
> 
> I’m going to close this case, but if you have more information regarding this issue, please reply to this email and the case will be re-opened for further investigation. We apologize for any inconvenience.
> 
> Best regards,
> 
> Claire
> 
> >

**Comment by vnckppl** [comment]
> > Hi Marcin,
> > Thanks for contacting Grammarly!
> > Unfortunately, we don’t currently have a quick fix, but please rest assured that our team has been made aware of the issue. I can’t provide you with an estimated timeframe right now because the developers need some time to analyze the cause of the issue.
> > I’m going to close this case, but if you have more information regarding this issue, please reply to this email and the case will be re-opened for further investigation. We apologize for any inconvenience.
> > Best regards,
> > Claire
> > >

Thanks for the update. That does not sound very promising, unfortunately.

**Comment by mmagnus** [comment]
Yep, unfortunately :(

**Comment by mmagnus** [comment]
https://github.com/jcs-elpa/grammarly 
https://github.com/jcs-elpa/flycheck-grammarly

but I'm not able to run it.

**Comment by mmagnus** [comment]
![demo](https://user-images.githubusercontent.com/118740/82449756-ff357500-9aab-11ea-96a9-4d853db44614.gif)
the workaround using Apple Scripts works, and it's bidirectional..

**Comment by mmagnus** [comment]
I also played a bit with sending the content to/from Safari and Grammarly, but nothing yet complete.

```
tell application "System Events"
	tell process "Safari"
		set frontmost to true
		tell application "Safari" to open location "https://app.grammarly.com/ddocs/771448507"
		delay 3
		click menu item "Paste" of menu "Edit" of menu bar 1
	end tell
end tell
```

your doc that is designed for Sandbox

BTW, for me, this Safari works faster than a stand-alone application. .hmm... maybe this is the way to go ...

**Comment by mmagnus** [comment]
OR THIS

https://github.com/jcs-elpa/flycheck-grammarly/

<img width="1970" alt="Screen Shot 2020-05-21 at 10 13 02 AM" src="https://user-images.githubusercontent.com/118740/82540412-4fb1de80-9b4f-11ea-8d6b-5f94dceb3fb9.png">

**Comment by vnckppl** [comment]
Hi mmagnus, 
Thanks for your work on this! I am able to use the scrips, but I do have to have Grammarly open with an empty document opened. Is this supposed to be the case?
Thanks!

**Comment by mmagnus** [comment]
@vnckppl  yeah, I could not get this running when the window is minimized. I'm missing something.. :(
